src/System.IO.Compression/tests/Performance/Perf.DeflateStream.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+using Microsoft.Xunit.Performance;
+using System.Collections.Generic;
+
+namespace System.IO.Compression.Tests
+{
+    public class Perf_DeflateStream
+    {
+        /// <summary>
+        /// Yields an Enumerable list of paths to GZTestData files
+        /// </summary>
+        public static IEnumerable<object[]> GZTestFiles()
+        {
+            yield return new object[] { "GZTestData/GZTestDocument.txt.gz" };
+            yield return new object[] { "GZTestData/GZTestDocument.docx.gz" };
+        }
+
+        [Benchmark]
+        [MemberData("GZTestFiles")]
+        public async void Decompress(string testFilePath)
+        {
+            int _bufferSize = 1024;
+            var bytes = new byte[_bufferSize];
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                int retCount = -1;
+                using (var gzStream = await LocalMemoryStream.readAppFileAsync(testFilePath))
+                using (var strippedMs = StripHeaderAndFooter.Strip(gzStream))
```

Comment:
DeflateStream disposes the stream from which it was created, so we cannot reuse the "strippedMs" stream.

The Strip function disposes the stream passed to it as input.